secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+struct Zatoshi {
+    enum Constants {
+        static let oneZecInZatoshi: Int64 = 100_000_000
+        static let maxZecs: Int64 = 21_000_000
+        static let maxZatoshi: Int64 = Constants.oneZecInZatoshi * Constants.maxZecs
+    }
+    
+    static var nsDecimalHandler = NSDecimalNumberHandler(
+        roundingMode: NSDecimalNumber.RoundingMode.bankers,
+        scale: 8,
+        raiseOnExactness: true,
+        raiseOnOverflow: true,
+        raiseOnUnderflow: true,
+        raiseOnDivideByZero: true
+    )
+    
+    @Clamped(-Constants.maxZatoshi...Constants.maxZatoshi)
+    var amount: Int64 = 0
+
+    /// Converts `Zatoshi` to `NSDecimalNumber`
+    var asZec: NSDecimalNumber {
```

Comment:
Since ZEC is not explicitly in the API I suggest we change it to something more descriptive

```suggestion
    var decimalValue: NSDecimalNumber {
```

Reply:
updated